# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/sandbox/chromium/base/memory/singleton.cc
# Commit: 370c40e49230
# Full Hash: 370c40e4923073a87845b57c1f4851bacd9663da
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2019-06-12 21:42:35
# Regressor Bug: 1552160
# File Overlap Count: 3
# Description:
#   Bug 1552160 Part 1: Roll-up of chromium sandbox update and mozilla patches to get a running browser. r=jld,aklotz,tjr,bobowen
#   
#   This updates security/sandbox/chromium/ files to chromium commit 84108231f6e6e0772fb9a4643679ce76aa771e67.
#   
#   Existing and new patches applied from security/sandbox/chromium-shim/patches/with_update/ to give a compiling and mostly working browser.
# ==============================================================================

diff -r bd5ecd72f76c -r 370c40e49230 security/sandbox/chromium/base/memory/singleton.cc
--- a/security/sandbox/chromium/base/memory/singleton.cc	Wed Jun 12 21:55:59 2019 +1200
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,34 +0,0 @@
-// Copyright (c) 2011 The Chromium Authors. All rights reserved.
-// Use of this source code is governed by a BSD-style license that can be
-// found in the LICENSE file.
-
-#include "base/memory/singleton.h"
-#include "base/threading/platform_thread.h"
-
-namespace base {
-namespace internal {
-
-subtle::AtomicWord WaitForInstance(subtle::AtomicWord* instance) {
-  // Handle the race. Another thread beat us and either:
-  // - Has the object in BeingCreated state
-  // - Already has the object created...
-  // We know value != NULL.  It could be kBeingCreatedMarker, or a valid ptr.
-  // Unless your constructor can be very time consuming, it is very unlikely
-  // to hit this race.  When it does, we just spin and yield the thread until
-  // the object has been created.
-  subtle::AtomicWord value;
-  while (true) {
-    // The load has acquire memory ordering as the thread which reads the
-    // instance pointer must acquire visibility over the associated data.
-    // The pairing Release_Store operation is in Singleton::get().
-    value = subtle::Acquire_Load(instance);
-    if (value != kBeingCreatedMarker)
-      break;
-    PlatformThread::YieldCurrentThread();
-  }
-  return value;
-}
-
-}  // namespace internal
-}  // namespace base
-